bundles/binding/org.openhab.binding.enphaseenergy/src/main/java/org/openhab/binding/enphaseenergy/internal/EnphaseenergyBinding.java

```diff
@@ -0,0 +1,185 @@
+     * {@inheritDoc}
+     */
+    @Override
+    protected long getRefreshInterval() {
+        return this.refreshInterval;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @SuppressWarnings("incomplete-switch")
+    @Override
+    protected void execute() {
+        logger.debug("Querying Enphase Energy API");
+		try {
+			// Find all systemId for active items 
+			Set<Integer>	systemIds = new HashSet<Integer>();
+			for (final EnphaseenergyBindingProvider provider : this.providers) {
+				for (final String itemName : provider.getItemNames()) {
+					final Integer systemId = provider.getSystemId(itemName);
+					systemIds.add(systemId);
+				}
+			}
+			for (final Integer systemId : systemIds) {
+				final SystemsRequest request = new SystemsRequest(this.key, this.user_id, systemId);
+				logger.debug("Request: {} as {}", request, request.prepare());
+
+				final SystemsResponse response = request.execute();
+				logger.debug("Response: {}", response);
+
+				for (final EnphaseenergyBindingProvider provider : this.providers) {
+					for (final String itemName : provider.getItemNames()) {
+						final Integer	itemSystemId = provider.getSystemId(itemName);
+						if(!systemId.equals(itemSystemId)) {
+							continue;
+						}
+						State state = null;
+						final EnphaseenergyItemType itemType = provider.getItemType(itemName);
+						logger.debug("itemName {} for {} and {}", itemName, systemId, itemType);
+						switch (itemType) {
+						case MODULES:
+							state = new DecimalType(response.getModules());
+							break;
+						case SIZE_W:
+							state = new DecimalType(response.getSize_w());
+							break;
+						case CURRENT_POWER:
+							state = new DecimalType(response.getCurrent_power());
+							break;
+						case ENERGY_TODAY:
+							state = new DecimalType(response.getEnergy_today() * 0.001);
+							break;
+						case ENERGY_LIFETIME:
+							state = new DecimalType(response.getEnergy_lifetime() * 0.001 * 0.001);
+							break;
+						case SUMMARY_DATE:
+							state = new DateTimeType(response.getSummary_date());
+							break;
+						case SOURCE:
+							state = new StringType(response.getSource());
+							break;
+						case STATUS:
+							state = new StringType(response.getStatus());
+							break;
+						case OPERATIONAL_AT:
+							state = new DateTimeType(response.getOperational_at());
+							break;
+						case LAST_REPORT_AT:
+							state = new DateTimeType(response.getLast_report_at());
+							break;
+						}
+						if (state != null) {
+							this.eventPublisher.postUpdate(itemName, state);
+						}
+					}
+				}
+            }
+		} catch (EnphaseenergyException ne) {
+			logger.error(ne.getMessage());
+		}
+	}
+    
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void updated(final Dictionary<String, ?> config) throws ConfigurationException {
+        if (config != null) {
+
+            Enumeration<String> configKeys = config.keys();
+            while (configKeys.hasMoreElements()) {
```

Comment:
since this binding seem to administer just a single account simple fetch properties like:

```
		if (config != null) {
			String hostnameString = (String) config.get("hostname");
			if (StringUtils.isNotBlank(hostnameString)) {
				hostname = hostnameString;
			}
			
			String refreshIntervalString = (String) config.get("refresh");
			if (StringUtils.isNotBlank(refreshIntervalString)) {
				refreshInterval = Long.parseLong(refreshIntervalString);
			}
			
			setProperlyConfigured(true);
		}
```